crates/index-scheduler/src/scheduler/mod.rs

```diff
@@ -221,7 +223,7 @@ impl IndexScheduler {
                     self.queue
                         .tasks
                         .update_task(&mut wtxn, &task)
-                        .map_err(|e| Error::TaskDatabaseUpdate(Box::new(e)))?;
+                        .map_err(|e| Error::TaskDatabaseUpgrade(Box::new(e)))?;
```

Comment:
consider removing the original `TaskDatabaseUpdate` and all its uses here.

Reply:
I went for opt 2